pkg/controller/elasticsearch/client/official_api.go

```diff
@@ -0,0 +1,482 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	elasticsearch6 "github.com/elastic/go-elasticsearch/v6"
+	esapi6 "github.com/elastic/go-elasticsearch/v6/esapi"
+	elasticsearch7 "github.com/elastic/go-elasticsearch/v7"
+	esapi7 "github.com/elastic/go-elasticsearch/v7/esapi"
+	elasticsearch8 "github.com/elastic/go-elasticsearch/v8"
+	esapi8 "github.com/elastic/go-elasticsearch/v8/esapi"
+	"github.com/elastic/go-elasticsearch/v8/esutil"
+)
+
+// putGenericClusterSettings is a generic helper method use to put cluster settings.
+func (c *elasticsearchClients) putGenericClusterSettings(
+	ctx context.Context,
+	settings io.Reader,
+	responseHandler func(response *Response) error,
+) error {
+	if err := c.doVersionedRequestWithUnversionedResponse(versionedRequestWithUnversionedResponse{
+		ES6: func(es *elasticsearch6.Client) (*esapi6.Response, error) {
+			return es.Cluster.PutSettings(
+				settings,
+				es.Cluster.PutSettings.WithContext(ctx),
+			)
+		},
+		ES7: func(es *elasticsearch7.Client) (*esapi7.Response, error) {
+			return es.Cluster.PutSettings(
+				settings,
+				es.Cluster.PutSettings.WithContext(ctx),
+			)
+		},
+		ES8: func(es *elasticsearch8.Client) (*esapi8.Response, error) {
+			return es.Cluster.PutSettings(
+				settings,
+				es.Cluster.PutSettings.WithContext(ctx),
+			)
+		},
+		ResponseHandler: responseHandler,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *elasticsearchClients) GetClusterHealth(ctx context.Context) (Health, error) {
+	var result Health
+
+	if err := c.doVersionedRequestWithUnversionedResponse(versionedRequestWithUnversionedResponse{
+		ES6: func(es *elasticsearch6.Client) (*esapi6.Response, error) {
+			return es.Cluster.Health(
+				es.Cluster.Health.WithContext(ctx),
+			)
+		},
+		ES7: func(es *elasticsearch7.Client) (*esapi7.Response, error) {
+			return es.Cluster.Health(
+				es.Cluster.Health.WithContext(ctx),
+			)
+		},
+		ES8: func(es *elasticsearch8.Client) (*esapi8.Response, error) {
+			return es.Cluster.Health(
+				es.Cluster.Health.WithContext(ctx),
+			)
+		},
+		ResponseHandler: esErrorOrDecodeJSON(&result),
+	}); err != nil {
+		return result, err
+	}
```

Comment:
So I guess an alternative to the `switch` in every call would be to have a `client` interface implemented by `clientv6`, `clientv7`, `clientv8`? Since they use different types, every single one would need to implement all functions of the generic client interface, and maybe panic or return an error if irrelevant (eg. zen1 on clientv7).
Which is pretty similar to the approach we had before this PR I think, but will be more verbose since need to implement all functions (cannot have clientv7 embed clientv6).

The major benefit I see is that we cannot accidentally forget a `case` in all switches: all clients will have to implement the interface. Also it avoids the `if versionX != nil` statements.

I don't think it should prevent this from being merged. Could be nice as a follow-up issue?